plans/init.pp

```diff
@@ -134,62 +134,21 @@
   }
 
   # Generate a parameters list to be fed to puppetlabs/peadm based on which
-  # architecture we've chosen to deploy. The default case should never be
-  # reached since any architecture has been previously validated.
-  case $architecture {
-    'xlarge': {
-      $base_params = {
-        'primary_host'            => $inventory['server'][0]['name'],
-        'primary_postgresql_host' => $inventory['psql'][0]['name'],
-        'compiler_hosts'          => $inventory['compiler'].map |$c| { $c['name'] },
-        'console_password'        => $console_password,
-        'dns_alt_names'           => [ 'puppet', $apply['pool']['value'] ],
-        'compiler_pool_address'   => $apply['pool']['value'],
-        'version'                 => $version
-      }
-      if $replica {
-        $params = merge($base_params, {
-          'replica_host'            => $inventory['server'][1]['name'],
-          'replica_postgresql_host' => $inventory['psql'][1]['name'],
-        })
-      } else {
-        $params = $base_params
-      }
-    }
-    'large': {
-      $base_params = {
-        'primary_host'           => $inventory['server'][0]['name'],
-        'compiler_hosts'        => $inventory['compiler'].map |$c| { $c['name'] },
-        'console_password'      => $console_password,
-        'dns_alt_names'         => [ 'puppet', $apply['pool']['value'] ],
-        'compiler_pool_address' => $apply['pool']['value'],
-        'version'               => $version
-      }
-      if $replica {
-        $params = merge($base_params, {
-          'replica_host' => $inventory['server'][1]['name'],
-        })
-      } else {
-        $params = $base_params
-      }
-    }
-    'standard': {
-      $base_params = {
-        'primary_host'           => $inventory['server'][0]['name'],
-        'console_password'      => $console_password,
-        'dns_alt_names'         => [ 'puppet', $apply['pool']['value'] ],
-        'compiler_pool_address' => $apply['pool']['value'],
-        'version'               => $version
-      }
-      if $replica {
-        $params = merge($base_params, {
-          'replica_host' => $inventory['server'][1]['name'],
-        })
-      } else {
-        $params = $base_params
-      }
-    }
-    default: { fail('Something went horribly wrong') }
+  # architecture we've chosen to deploy. PEAdm will figure out the correct
+  # thing to do based on whether or not there are valid values for each
+  # architecture component. An empty array is equivalent to not defining the
+  # parameter.
+  $params = {
+    'primary_host'            => [getvar('inventory.server.0.name')].autope::compact(),
+    'primary_postgresql_host' => [getvar('inventory.psql.0.name')].autope::compact(),
+    'compiler_hosts'          => [getvar('inventory.compiler')].autope::compact().map |$c| { $c['name'] },
```

Comment:
It seems this line will complain about: 

`"Evaluation Error: The value 'name' cannot be converted to Numeric. (file: /var/folders/37/42xxy3tj1p7dv7ptq0vk5sm40000gn/T/puppet_debugger_input20210719-49835-5xyeoc.pp, line: 1, column: 62)"
`
```suggestion
    'compiler_hosts'          => getvar('inventory.compiler').autope::compact().map |$c| { $c['name'] },
```